camera_system/block_to_intersection.py

```diff
@@ -12,15 +12,23 @@
 class BlockToIntersection(GameMotion):
     """ブロック置き場→交点のゲーム動作クラス."""
 
-    def __init__(self, angle: int, target_color: Color) -> None:
+    def __init__(self, angle: int, target_color: Color, have_block: bool) -> None:
         """BlockToIntersectionのコンストラクタ.
 
         Args:
             angle: 方向転換の角度
             target_color: 目標となる交点の色
-
+            have_block: ブロックを保持している場合True
         """
-        self.__angle = angle
+        if have_block:  # ブロックを保持している場合
+            self.__setting_angle = GameMotion.ROTATION_BLOCK_TABLE[abs(angle)]["angle"]
+            self.__rotation_pwm = GameMotion.ROTATION_BLOCK_PWM
+            self.__rotation_time = GameMotion.ROTATION_BLOCK_TABLE[abs(angle)]["time"]
+        else:  # ブロックを保持していない場合
+            self.__setting_angle = GameMotion.ROTATION_NO_BLOCK_TABLE[abs(angle)]["angle"]
+            self.__rotation_pwm = GameMotion.ROTATION_NO_BLOCK_PWM
+            self.__rotation_time = GameMotion.ROTATION_NO_BLOCK_TABLE[abs(angle)]["time"]
+        self.__clockwise = "clockwise" if angle > 0 else "anticlockwise"
```

Comment:
```self.__clockwise = "anticlockwise" ```はおかしい気がするので
```suggestion
        self.__direct_rotation = "clockwise" if angle > 0 else "anticlockwise"
```
とかの方がいい気がします。